### DISCUSSION TITLE
How to make truncate responsive?

### DISCUSSION THREAD

**Discussion by polkabrothers** [tailwindlabs/tailwindcss#3509] [Help] [answered]
Is it possible to make .truncate responsive?

I'd like to truncate text on smaller devices, but display the full text on larger screens.
Something like `<div class="truncate lg:truncate-none">`

**Comment by simonswiss** [comment, marked as answer]
Hey!

the `truncate` class sets three properties:

```css
.truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}
```

The `overflow` and `white-space` properties can be set individually as well with `overflow-{value}` and `whitespace-{value}` utilities, so it's possible to undo a truncation at a larger screen size by applying a `whitespace-normal` class at the breakpoint where you want to "undo" the text truncation

```html
<p class="truncate md:whitespace-normal">I will be only truncated up to the `md` breakpoint, and then fully visible on larger screens!</p>
```

Here's a Tailwind Play example to show it in action:

https://play.tailwindcss.com/Lo7uSPMQmk

Hope it helps!

**Reply by Saadi-Rahman** [reply]
Thanks a lot.

**Reply by y-nk** [reply]
thanks for the solution, but this is sad. `truncate-none` seems a good addition.
